src/unread/unreadHelpers.js

```diff
@@ -7,19 +7,12 @@ type SomeUnreadItem = { unread_message_ids: number[] };
 export function removeItemsDeeply<T: SomeUnreadItem>(objArray: T[], messageIds: number[]): T[] {
   let changed = false;
   const objWithAddedUnreadIds = objArray.map(obj => {
-    const newIds = removeItemsFromArray(obj.unread_message_ids, messageIds);
-    if (newIds.length === obj.unread_message_ids) {
```

Comment:
:-/ Yeah.

One thing is that this isn't *unsound* -- it's just always false. So it should be basically a lint-style warning.

And then one tricky thing, I guess, is that it's often intentional to have a check that the types say should always be false -- because there's some un-type-checked code elsewhere in the system and you're not sure if it behaves properly. Ideally you'd express that uncertainty in the types of that code's interfaces, so that you'd necessarily check at the boundary and then the type-checker would guarantee you'd adequately checked, but that's often not how it works out. So I think I'd really want a way to silence that lint warning explicitly on a given check, with a hammer smaller than `$FlowFixMe` that won't also suppress unrelated actual type errors if they arise.

Still, none of that seems particularly insurmountable, and I agree it'd be a good check to have.